### PR TITLE
edit behaviour for employee object

### PR DESCRIPTION
**Pull request by jgj-magenta-aps**
- [X] Have you updated the documentation?
- [X] Have you performed a smoke test of the application?
- [X] Have you written tests for your changes?
- [X] Have you updated the release notes?

### CODE REVIEW

File: backend/tests/test_integration_employee.py [hunk 1]
```diff
@@ -458,3 +458,180 @@ def test_cpr_lookup_raises_on_wrong_length(self):
             },
             status_code=400,
         )
+
+    def test_edit_employee_overwrite(self):
```
Comment (cmoesgaard) [inline backend/tests/test_integration_employee.py]: This test should assert that `cpr_no` has been changed correctly as well.

File: backend/tests/test_integration_employee.py [hunk 2]
```diff
@@ -458,3 +458,180 @@ def test_cpr_lookup_raises_on_wrong_length(self):
...
+        self.assertEqual(
+            expected_brugeregenskaber,
+            actual['attributter']['brugeregenskaber']
+        )
+        self.assertEqual(
+            expected_brugergyldighed,
+            actual['tilstande']['brugergyldighed']
+        )
+
+    def test_edit_employee(self):
```
Comment (cmoesgaard) [inline backend/tests/test_integration_employee.py]: Ditto.

File: backend/mora/mapping.py [hunk 3]
```diff
@@ -66,10 +68,10 @@
 
 # Org unit
 ORG_UNIT_TYPE = 'org_unit_type'
-NAME = 'name'
+# NAME = 'name' - see under common
 PARENT = 'parent'
 ADDRESSES = 'addresses'
-LOCATION = 'location'
+# LOCATION = 'location' - see under common
```
Comment (danchr) [inline backend/mora/mapping.py]: Ditto.

File: backend/mora/service/employee.py [hunk 4]
```diff
@@ -35,6 +35,200 @@
                             url_prefix='/service')
 
 
+class BrugerRequestHandler(handlers.RequestHandler):
+    __slots__ = ['details_requests']
```
Comment (danchr) [inline backend/mora/service/employee.py]: I know this is a total nit, but `__slots__` is usually a tuple not a list. Although unlikely to matter in practice, tuples are unmodifiable:
```suggestion
    __slots__ = ('details_requests',)
```

File: backend/mora/service/employee.py [hunk 5]
```diff
@@ -35,6 +35,200 @@
...
+    role_type = "employee"
+
+    def prepare_create(self, req):
+
```
Comment (danchr) [inline backend/mora/service/employee.py]: Extraneous newline.

File: backend/mora/service/employee.py [hunk 6]
```diff
@@ -35,6 +35,200 @@
...
+        """Create a new employee
```
Comment (danchr) [inline backend/mora/service/employee.py]: Although it probably out to be, this isn't actually where we get the documentation. Rather, it's extracted from `detail_writing` — which you also provide. Any docstring here should be more about the code. At least for now :/

(Note that the other classes don't actually provide one.)

File: backend/mora/service/detail_writing.py [hunk 7]
```diff
@@ -393,6 +393,47 @@ def edit():
...
+    The **original** and **data** objects follow the same structure.
+    Every field in **original** is required, whereas **data** only needs
+    to contain the fields that need to change along with the validity dates.
+
+    :<jsonarr string cpr_no: The cpr number of the employee
+    :<jsonarr string name: The Name of the employee
+    :<jsonarr object validity: The validities of the changes.
+
+    .. sourcecode:: json
+        [{
```
Comment (danchr) [inline backend/mora/service/detail_writing.py]: In order for Sphinx to pick up the directory, you need a newline between the directive and the text. Then, you'll get a warning about your JSON not being actually valid JSON. (`None` ≠ `null`; no final commas; missing quotes.)

To build the docs, use `./flask.sh docs`.

File: backend/tests/test_integration_employee.py (hunk 2 above)
Comment (jgj-magenta-aps) [reply backend/tests/test_integration_employee.py]: ditto-ditto fixed

File: backend/mora/service/employee.py [hunk 8]
```diff
@@ -35,6 +35,200 @@
                             url_prefix='/service')
 
 
+class BrugerRequestHandler(handlers.RequestHandler):
```
[the remark this answers is not in the document]
Comment (jgj-magenta-aps) [reply backend/mora/service/employee.py]: Yes - and done

File: backend/mora/mapping.py (hunk 3 above)
Comment (jgj-magenta-aps) [reply backend/mora/mapping.py]: removed are now really removed

File: backend/mora/service/employee.py (hunk 4 above)
Comment (jgj-magenta-aps) [reply backend/mora/service/employee.py]: yup

File: backend/mora/service/employee.py (hunk 6 above)
Comment (jgj-magenta-aps) [reply backend/mora/service/employee.py]: I had skipped this exercise - I only had the detail_writing-edit part documented.
I have now moved the create-docu there too and outfitted it with a "type": "employee"

**Pull request merged**